lib/vector-core/src/event/proto.rs

```diff
@@ -41,6 +47,18 @@ impl From<Log> for event::LogEvent {
     }
 }
 
+impl From<Trace> for event::LogEvent {
+    fn from(trace: Trace) -> Self {
+        let fields = trace
+            .fields
+            .into_iter()
+            .filter_map(|(k, v)| decode_value(v).map(|value| (k, value)))
+            .collect::<BTreeMap<_, _>>();
+
+        Self::from(fields)
+    }
+}
```

Comment:
I'm curious where this is useful without also setting up the `message` field, etc.

Reply:
This is needed because of  https://github.com/vectordotdev/vector/blob/c78a1d670b6ace44e2614f6256c273a20d54ed5b/lib/vector-core/src/event/proto.rs#L144 for conversion from protobuf (hence the dump-all-fields-logic).